gui/src/main/index.ts

```diff
@@ -582,10 +582,10 @@ class ApplicationMain {
     }
   }
 
-  private setTunnelState(newState: TunnelState) {
+  private async setTunnelState(newState: TunnelState) {
     this.tunnelState = newState;
     this.updateTrayIcon(newState, this.settings.blockWhenDisconnected);
-    this.updateLocation();
+    await this.updateLocation();
```

Comment:
This call may take up to 10 seconds to complete as it calls `daemonRpc.getLocation()`. We should never truly wait for it to complete. It may as well fail.